src/app/(profile)/profiles/[username]/page.tsx

```diff
@@ -44,11 +47,22 @@ const UserProfilePage = async ({
   params: { username: string }
 }) => {
   const selectedProfile = await getProfileByGithubUsername(params.username)
+  const session = await getServerSession(authOptions)
 
   if (!selectedProfile) {
     redirect(AppRoutes.profiles)
   }
 
+  if (!session?.user) {
```

Comment:
**issue:** we should not add this redirect, as currently previewing profiles is open for public